File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/files/ListFiles.java
package com.pubnub.api.java.endpoints.files;

import com.pubnub.api.java.endpoints.BuilderSteps;
import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.models.consumer.files.PNListFilesResult;

public interface ListFiles extends Endpoint<PNListFilesResult> {
    ListFiles limit(Integer limit);

    ListFiles next(com.pubnub.api.models.consumer.objects.PNPage.PNNext next);

    interface Builder extends BuilderSteps.ChannelStep<ListFiles> { }
}

File: pubnub-kotlin/pubnub-kotlin-api/src/commonMain/kotlin/com/pubnub/api/endpoints/files/ListFiles.kt
package com.pubnub.api.endpoints.files

import com.pubnub.api.models.consumer.files.PNListFilesResult
import com.pubnub.kmp.PNFuture

/**
 * @see [PubNub.listFiles]
 */
expect interface ListFiles : PNFuture<PNListFilesResult>
